src/sagemaker/session.py

```diff
@@ -4964,6 +4973,17 @@ def create_inference_recommendations_job(
             str: The name of the job created. In the form of `SMPYTHONSDK-<timestamp>`
         """
 
+        if model_name is None and model_package_version_arn is None:
+            raise ValueError(
+                "Missing model_name and model_package_version_arn," " please provide one of them."
+            )
+
+        if model_name is not None and model_package_version_arn is not None:
+            raise ValueError(
+                "Please provide either model_name or model_package_version_arn"
```

Comment:
nit: think u combined two sentences here? "Please provide either model_name or model_package_version_arn." or "Either model_name or model_package_version_arn should be provided, not both."